src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -36,12 +35,25 @@ public static IReadOnlyList<SemanticRange> VisitAllNodes(RazorCodeDocument razor
             return visitor._semanticRanges;
         }
 
+        private void Visit(SyntaxList<RazorSyntaxNode> syntaxNodes)
+        {
+            for (var i = 0; i < syntaxNodes.Count; i++)
+            {
+                Visit(syntaxNodes[i]);
+            }
+        }
+
         #region HTML
         public override void VisitMarkupAttributeBlock(MarkupAttributeBlockSyntax node)
         {
+            Visit(node.NamePrefix);
             AddSemanticRange(node.Name, RazorSemanticTokensLegend.MarkupAttribute);
+            Visit(node.NameSuffix);
             AddSemanticRange(node.EqualsToken, RazorSemanticTokensLegend.MarkupOperator);
-            base.VisitMarkupAttributeBlock(node);
```

Comment:
While modifying this class I ran into an instance where I was visiting a node infinitely because I did all of the above and then called the base method. Since the existing formulation is technically wrong and sets us up for a trap I just went through and started visiting each of the nodes in our overloads manually. Most of these (the prefix and suffix in particular) are unlikely to do anything, but they're more correct, keep us from infinitely recursing, and should make any future functionality additions easier.

Reply:
Sounds fine to me